### PR TITLE
chore: Fire an error when StorageClass/PersistentVolume uses in-tree Storage provider

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes # 

**Description**

Fire an error when StorageClass uses in-tree storage driver. This PR should surface errors when PVCs are using the in-tree storage driver vs. the CSI driver. In this case, Karpenter isn't able to discover the volume allocation limit and will not spin up new nodes if the limit is passed.

Most of this PR is some moves and some additional commenting for documentation on what these functions are doing

**How was this change tested?**

`make apply`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/scheduling/volumeusage.go [hunk 1]
```diff
@@ -97,37 +139,6 @@ func (v *VolumeUsage) Add(ctx context.Context, kubeClient client.Client, pod *v1
 	v.volumes = v.volumes.union(podVolumes)
 }
 
-type VolumeCount map[string]int
```
Comment (tzneal) [inline pkg/scheduling/volumeusage.go:100]: Can you undo the file re-organization? It makes the diff larger and more difficult to review.

File: pkg/scheduling/volumeusage.go [hunk 2]
```diff
@@ -187,14 +185,51 @@ func (v *VolumeUsage) validate(ctx context.Context, kubeClient client.Client, po
...
+func (v *VolumeUsage) driverFromSC(ctx context.Context, kubeClient client.Client, storageClassName string) (string, error) {
 	var sc storagev1.StorageClass
-	if err := kubeClient.Get(ctx, client.ObjectKey{Name: *storageClassName}, &sc); err != nil {
+	if err := kubeClient.Get(ctx, client.ObjectKey{Name: storageClassName}, &sc); err != nil {
 		return "", err
 	}
+	// Check if the provisioner name is an in-tree plugin name
+	if csiName, err := translator.GetCSINameFromInTreeName(sc.Provisioner); err == nil {
+		if changeMonitor.HasChanged(storageClassName, nil) {
+			logging.FromContext(ctx).With("storage-class", storageClassName, "provisioner", sc.Provisioner).Errorf("StorageClass .spec.provisioner uses an in-tree storage plugin which is unsupported by Karpenter. Scale-ups may fail because Karpenter will not discover driver limits. Create a new StorageClass with a .spec.provisioner referncing the CSI driver plugin name '%s'", csiName)
```
Comment (tzneal) [inline pkg/scheduling/volumeusage.go:226]: typo in `referencing`

File: pkg/scheduling/volumeusage.go (hunk 1 above)
Comment (jonathan-innis) [reply pkg/scheduling/volumeusage.go:100]: Updated

**Review by tzneal — DISMISSED** [review]
lgtm

**Review by tzneal — APPROVED** [review]
lgtm

**Pull request merged**